homeassistant/helpers/template.py

```diff
@@ -493,6 +494,7 @@ async def async_render_will_timeout(
         finish_event = asyncio.Event()
 
         def _render_template() -> None:
+            assert self.hass is not None
```

Comment:
If we can't remove the assert, we should add a message
```suggestion
            assert self.hass is not None, "hass variable not set on template"
```